# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1528305200 14400
#      Wed Jun 06 13:13:20 2018 -0400
# Node ID 5523d88bb5d39a3a2836853fe3a010a0943b3fe7
# Parent  996d78102a719cdfe28fbd68f24adb9874eb143a
* open.m: Check arg type before checking file type.

diff --git a/scripts/miscellaneous/open.m b/scripts/miscellaneous/open.m
--- a/scripts/miscellaneous/open.m
+++ b/scripts/miscellaneous/open.m
@@ -56,24 +56,26 @@
 ##
 ## Other file types are opened in the appropriate external application.
 ## @end deftypefn
 
 function output = open (file)
 
   if (nargin != 1)
     print_usage ();
-  elseif (! exist (file, "file"))
-    error ("open: unable to find file %s", file);
   endif
 
   if (! ischar (file))
     error ("open: FILE must be a string");
   endif
 
+  if (! exist (file, "file"))
+    error ("open: unable to find file %s", file);
+  endif
+
   [~, fname, ext] = fileparts (file);
 
   if (! isempty (ext)
       && any (exist (["open" tolower(ext(2:end))]) == [2 3 5 103]))
     try
       feval (["open" tolower(ext(2:end))], file)
     catch
       error ("open: %s", lasterr);
